compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -2977,6 +3150,39 @@ fn eval_bin_op_with_bool_literals(
     Value::Bool(bin_op_result)
 }
 
+fn eval_bin_op_with_double_literals(
+    bin_op: BinOp,
+    lhs_literal: Literal,
+    rhs_literal: Literal,
+    bin_op_expr_span: PackageSpan, // For diagnostic purposes only
+) -> Result<Value, Error> {
+    fn eval_double_div(lhs: f64, rhs: f64, span: PackageSpan) -> Result<Value, Error> {
+        match (lhs, rhs) {
+            (_, 0.0) => Err(EvalError::DivZero(span).into()),
+            (lhs, rhs) => Ok(Value::Double(lhs / rhs)),
+        }
+    }
+
+    // Validate that both literals are doubles.
+    let (Literal::Double(lhs), Literal::Double(rhs)) = (lhs_literal, rhs_literal) else {
+        panic!("at least one literal is not an double: {lhs_literal}, {rhs_literal}");
+    };
+
+    match bin_op {
+        BinOp::Eq => Ok(Value::Bool(f64_approx_eq(lhs, rhs))),
+        BinOp::Neq => Ok(Value::Bool(f64_approx_neq(lhs, rhs))),
+        BinOp::Gt => Ok(Value::Bool(lhs > rhs)),
+        BinOp::Gte => Ok(Value::Bool(f64_approx_gte(lhs, rhs))),
+        BinOp::Lt => Ok(Value::Bool(lhs < rhs)),
```

Comment:
It occurred to me last night that this is still problematic. For example, if rhs is greater than lhs but within epsilon, then both "rhs > lhs" and "rhs <= lhs" will be true when they should be mutually exclusive (indeed, "<=" if often defined in terms of "not >").

I have some thoughts on what a 'correct' impl would look like, but do you know if there is any prior art we should follow here in other languages re handling floats to within an epsilon?

Reply:
It is all removed.